src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts

```diff
@@ -89,6 +90,26 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
     this.rateModelDictionary = new across.rateModel.RateModelDictionary();
   }
 
+  /**
+   * Retrieves the most recently set UBA config for a given L1 token address before a block number.
+   * @param l1TokenAddress The L1 token address to retrieve the config for
+   * @param blockNumber The block number to retrieve the config for. If not specified, sets block to max integer
+   * meaning that this function will return the latest config.
+   * @returns The UBA config for the given L1 token address and block number, or undefined if no config exists
+   * before blockNumber.
+   */
+  getUBAConfig(l1TokenAddress: string, blockNumber = Number.MAX_SAFE_INTEGER): UBAParsedConfigType | undefined {
+    const config = findLast(
+      this.ubaConfigUpdates,
+      (config) => config.l1Token === l1TokenAddress && config.blockNumber <= blockNumber
+    );
+    return config?.config;
+  }
```

Comment:
Should the UBA specific functions reside in the UBAClient utilities?

Reply:
I don't see this as a UBA specific function, rather I see it as a function to fetch and parse the on chain `TokenConfig` event emitted by the `AcrossConfigStore` and return it as a usable object. the `UBAFeeConfig` then takes the result of this `ConfigStoreClient.getUBAConfig` call and shapes it further as an opinionated `UBAFeeConfig` that can be used by the `UBAClient`

Reply:
SGTM